src/server/req_signal.c

```diff
@@ -113,12 +113,17 @@ req_signaljob(struct batch_request *preq)
 	int		  resume = 0;
 	char		 *vrange;
 	int		  x, y, z;
+	int 		 err = PBSE_NONE;
 
 	jid = preq->rq_ind.rq_signal.rq_jid;
 
-	parent = chk_job_request(jid, preq, &jt);
-	if (parent == NULL)
-		return;		/* note, req_reject already called */
+	parent = chk_job_request(jid, preq, &jt, &err);
+	if (parent == NULL) {
+		pjob = find_job(jid);
+		if (pjob != NULL && pjob->ji_pmt_preq != NULL)
+			reply_preempt_jobs_request(err, PREEMPT_METHOD_SUSPEND, pjob);
```

Comment:
instead of doing this for every caller, wouldn't it be good to move this to chk_job_request() itself?

Reply:
Originally I thought chk_job_request() was just used for deletion.  It turns out it is called all over the place.  So my option was to pass in the preempt method, or to do it in the caller.  Since there are 3 calls which use the preempt method and about 10 without, I went this way.  I also thought it might be useful for this function to return the reason why it failed for the future.